lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -940,7 +940,10 @@ DotOperandEncodingAttr::getElemsPerThread(ArrayRef<int64_t> shape,
   } else if (auto mma = mlir::dyn_cast<NvidiaMmaEncodingAttr>(parent)) {
     if (mma.isAmpere() || mma.isHopper()) {
       auto bitwidth = getPointeeType(eltTy).getIntOrFloatBitWidth();
-      auto rep = mma.getRepForOperand(shape, bitwidth, idx);
+      // On hopper, the number of elements per thread is independent on the type
+      // and will only depend on kWidth.
+      int repBitwidth = mma.isHopper() ? 32 / kWidth : bitwidth;
+      auto rep = mma.getRepForOperand(shape, repBitwidth, idx);
```

Comment:
Shouldn't the fix go in `NvidiaMmaEndcodingAttr::getRepForOperand`? This seems a bit hacky.

Reply:
The Rep implementation is correct. `getRepForOperand` returns the number of times that we repeat the tile. Then, here we should multiply by the number of elements in each tile, which is given by `getSizePerThread()`. We should do this for both Ampere and Hopper.

Reply:
yes I believe we should use kWidth for both Ampere and Hopper. It will require more changes for Ampere though

Reply:
IIUC the bitwidth is used by `getRepForOperand` to compute the number of elements in the core matrix, but it's incorrect for hopper so `getRepForOperand` should use 
```
auto shapePerWarp[] = {1, 16, 8, isHopper() ? 4 * 2 * getKWidth() : 4 * 64 / bitwidth};
```

Reply:
fair enough, changed it to what Peter suggested. 
Note that the kWidth is a bit of a confusing term for hopper. At some point we should refactor it